src/emmc.c

```diff
@@ -148,15 +148,14 @@ static gboolean r_emmc_force_part_write(const gchar *device, gchar value, GError
 	g_return_val_if_fail(value == '0' || value == '1', FALSE);
 
 	sysfs_path = g_strdup_printf("/sys/block/%s/force_ro", device_basename);
-	if (!g_file_test(sysfs_path, G_FILE_TEST_EXISTS)) {
+	f = g_fopen(sysfs_path, "w");
+	if (!f) {
 		g_set_error(error, R_UPDATE_ERROR, R_UPDATE_ERROR_FAILED,
-				"Block device does not have force_ro attribute: %s does not exist",
-				sysfs_path);
+				"Could not open device attribute %s: %s",
+				sysfs_path, g_strerror(errno));
```

Comment:
Please save `errno` with `int err = errno;` as suggested by glib documentation and as we do for other similar calls, too.

Reply:
I've adjusted the other calls in `emmc.c` as well.